src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -129,6 +132,11 @@ private Task SetRoutingAndContinue(HttpContext httpContext)
                 _metrics.MatchSuccess(route, isFallback);
             }
 
+            // Map RequestSizeLimitMetadata to IHttpMaxRequestBodySizeFeature if present on the endpoint.
+            // We do this during endpoint routing to ensure that successive middlewares in the pipeline
+            // can access the feature with the correct value.
+            SetMaxRequestBodySize(httpContext);
```

Comment:
This seems unrelated to antiforgery?

Reply:
Kinda sorta.

Earlier in .NET 8, we added support for the concept of `IRequestBodySizeMetadata` for endpoints. Prior to this PR, it's only respected in two places:

- In MVC's `RequestBodySizeAttribute` where an MVC filter reads from the metadata and sets the `IHttpMaxRequestBodySizeFeature`
- In the new RequestDecompressionMiddleware which uses it to ensure that limit is requested when the request body is compressed

Because we're doing the anti-forgery check in middleware and because the check requires reading from the request body to extract the anti-forgery token, we need to make sure that the request size limit is set as early as possible in the request pipeline so that we respect it when we read the body.

This is also generally the right thing to do because it means that all middlewares that can potentially read from the body will be reading it at a point when the body request feature is set correctly. It just so happens that our built-in anti-forgery middleware is the first one that does this.

Reply:
OK that makes sense thanks.